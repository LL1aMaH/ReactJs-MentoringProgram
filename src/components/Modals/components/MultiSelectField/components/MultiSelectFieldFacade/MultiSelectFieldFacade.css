.layout {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.3125rem 4rem 0.3125rem 10px;
  position: relative;
  border-radius: 4px;
  color: var(--text-secondary-color);
  cursor: pointer;
  outline: none;
  user-select: none;
  background-color: var(--primary-contrast-color);
  border: none;
  width: 100%;
  height: 3rem;
  font-size: 15px;
}

.value {
  max-width: 100%;
  display: flex;
  align-items: center;
}

.icons {
  width: 2rem;
  height: auto;
  display: flex;
  align-self: center;
  position: absolute;
  right: 0.5rem;
  margin: 0 auto;
}

.icon {
  outline: none;

  &:last-child {
    margin: 0 auto;
    margin-left: 0.5rem;
  }
}

.active {
  border: 1px solid var(--primary-link-color);
}

.activeArrow {
  transform: rotate(180deg);
}
