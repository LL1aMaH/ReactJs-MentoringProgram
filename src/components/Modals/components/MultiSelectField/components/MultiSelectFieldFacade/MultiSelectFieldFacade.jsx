import React, { memo } from 'react';
import classnames from 'classnames';

import ArrowIcon from '../../../../../../../public/dropdown.svg'
'Assets/icons/dropdown.svg';

import styles from '../../../../../../../styles/MultiSelectFieldFacade.module.scss'

export const MultiSelectFieldFacade = memo(function MultiSelectFieldFacade(
  props,
) {
  const { children, onClick, active = false } = props;

  return (
    <div
      className={classnames(styles.layout, { [styles.active]: active })}
      onClick={onClick}
      role="presentation"
      data-testid="multi-select-field-facade"
    >
      <div className={styles.value}>{children}</div>
      <div className={styles.icons}>
        <ArrowIcon
          className={classnames(styles.icon, { [styles.activeArrow]: active })}
          tabIndex={-1}
          data-testid="arrow-icon"
        />
      </div>
    </div>
  );
});
