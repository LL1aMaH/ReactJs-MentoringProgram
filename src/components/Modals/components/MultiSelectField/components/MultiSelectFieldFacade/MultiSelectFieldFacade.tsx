import React, { memo, ReactNode } from 'react';
import classnames from 'classnames';

import ArrowIcon from 'Assets/icons/dropdown.svg';

import styles from './MultiSelectFieldFacade.css';

export type TMultiSelectFieldFacadeProps = {
  children?: ReactNode;
  onClick: () => void;
  active?: boolean;
};

export const MultiSelectFieldFacade = memo(function MultiSelectFieldFacade(
  props: TMultiSelectFieldFacadeProps,
): JSX.Element {
  const { children, onClick, active = false } = props;

  return (
    <div
      className={classnames(styles.layout, { [styles.active]: active })}
      onClick={onClick}
      role="presentation"
      data-testid="multi-select-field-facade"
    >
      <div className={styles.value}>{children}</div>
      <div className={styles.icons}>
        <ArrowIcon
          className={classnames(styles.icon, { [styles.activeArrow]: active })}
          tabIndex={-1}
          data-testid="arrow-icon"
        />
      </div>
    </div>
  );
});
