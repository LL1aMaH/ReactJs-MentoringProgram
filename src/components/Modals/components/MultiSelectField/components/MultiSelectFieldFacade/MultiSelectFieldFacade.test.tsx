import React from 'react';
import { render } from '@testing-library/react';

import { MultiSelectFieldFacade } from './MultiSelectFieldFacade';

describe('MultiSelectFieldFacade', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MultiSelectFieldFacade onClick={jest.fn()}>children</MultiSelectFieldFacade>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
