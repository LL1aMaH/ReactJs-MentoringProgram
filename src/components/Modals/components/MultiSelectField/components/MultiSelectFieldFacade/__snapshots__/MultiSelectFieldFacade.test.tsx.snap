// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectFieldFacade should render correctly 1`] = `
<DocumentFragment>
  <div
    class="layout"
    data-testid="multi-select-field-facade"
    role="presentation"
  >
    <div
      class="value"
    >
      children
    </div>
    <div
      class="icons"
    >
      <svg
        class="icon"
        data-testid="arrow-icon"
        tabindex="-1"
      />
    </div>
  </div>
</DocumentFragment>
`;
